x-pack/platform/plugins/shared/stack_connectors/server/connector_types/resilient/utils.ts

```diff
@@ -5,51 +5,127 @@
  * 2.0.
  */
 
-import { isArray } from 'lodash';
-import type { GetValueTextContentResponse, UpdateIncidentRequest } from './types';
-
-export const getValueTextContent = (
-  field: string,
-  value: string | number | number[]
-): GetValueTextContentResponse => {
-  if (field === 'description') {
-    return {
-      textarea: {
-        format: 'html',
-        content: value.toString(),
-      },
-    };
+import { isNumber, isString, isBoolean, isObject } from 'lodash';
+import type {
+  ResilientUpdateFieldValue,
+  ResilientTextAreaField,
+  ResilientFieldPrimitives,
+  UpdateIncidentRequest,
+  CreateIncidentData,
+  Incident,
+} from './types';
+import type { ResilientFieldMeta } from './schema';
+
+const getValueFromOldField = (
+  fieldMeta: ResilientFieldMeta,
+  value: unknown
+): ResilientFieldPrimitives => {
+  switch (fieldMeta.input_type) {
+    case 'textarea':
+      return value === null || value === undefined
+        ? null
+        : (value as ResilientTextAreaField['textarea']);
+    case 'multiselect':
+    case 'select':
+    case 'datetimepicker':
+    case 'datepicker':
+    case 'boolean':
+    case 'number':
+    case 'text':
+      return value as ResilientFieldPrimitives;
+    default:
+      return null;
   }
+};
 
-  if (field === 'incidentTypes') {
-    if (isArray(value)) {
-      return { ids: value.map((item) => Number(item)) };
+export function validateValues(
+  fieldMeta: ResilientFieldMeta,
+  givenValues: ResilientFieldPrimitives[]
+) {
+  const valuesMeta = fieldMeta.values;
+  if (valuesMeta && Array.isArray(valuesMeta)) {
+    if (!givenValues.every((id) => valuesMeta.some((option) => option.value === id))) {
+      throw new Error(
+        `Invalid values provided to ${fieldMeta.name}: ${JSON.stringify(
+          givenValues,
+          null,
+          2
+        )}. Accepted values: ${valuesMeta.map((v) => `${v.value} (for "${v.label}")`)}`
```

Comment:
Really nice error messaging here to get around the multi-select expectations. Thanks!